fil_fungible_token/src/token/mod.rs

```diff
@@ -118,7 +123,7 @@ where
         amount: &TokenAmount,
         operator_data: RawBytes,
         token_data: RawBytes,
-    ) -> Result<TokensReceivedParams> {
+    ) -> Result<ReceiverHookGuard<TokenAmount>> {
```

Comment:
I think this `TokenAmount` (total supply) should be either wrapped in a struct again, or documented in the comments.

Reply:
I have landed a `MintReturn` on master already